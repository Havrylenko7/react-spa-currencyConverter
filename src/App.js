import React from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import { Routes, Route } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';

import { getCurrencyFetch } from './redux/ducks/currency';
import { Navigation, Converter, Rates } from './components';
import Grid from './styled/App';

const App = () => {  
  const dispatch = useDispatch();
  const rates = useSelector(state => state.currencyReducer.rates);

  React.useEffect(() => {
    dispatch(getCurrencyFetch())
  }, [])

  if (rates.eur === undefined) {
    return <div className="loading">Loading...</div>;
  } else {
    return (
      <Grid>
        <Navigation />
        <Routes>  
          <Route path="/rates" element={<Rates />}/>
          <Route path="/converter" element={<Converter />}/>
        </Routes> 
      </Grid>
    );
  }
}

export default App;
