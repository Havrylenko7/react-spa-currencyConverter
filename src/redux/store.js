import { combineReducers, createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from '@redux-saga/core';

import { currencyReducer, saga } from './ducks/currency';

const sagaMiddleware = createSagaMiddleware();
const rootReducer = combineReducers({ currencyReducer });
const store = createStore(rootReducer, applyMiddleware(sagaMiddleware));

sagaMiddleware.run(saga);

export default store;
