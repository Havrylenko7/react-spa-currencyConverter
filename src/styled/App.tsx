import styled from 'styled-components';

import { color, media } from './Global';

const Grid = styled.div`
  display: grid;
  font-size: 18px;
  font-family: Fantasy;
  max-height: 100%;
  grid-gap: 20px;
  padding: 20px;
  margin: 7.5% 30%;
  border-radius: 10px;
  background-color: ${color.blue};
  box-shadow: 0 10px 25px ${color.black};
  
    @media (max-width: ${media.tabletWidth}) {
      margin: 5% 20%;
    }
    @media (max-width: ${media.phoneWidth}) {
      margin: 2.5% 15%;
  }
`;

export default Grid;
