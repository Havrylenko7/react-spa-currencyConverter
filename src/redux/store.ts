import { combineReducers, createStore, applyMiddleware } from 'redux';
import createSagaMiddleware from '@redux-saga/core';
import { TypedUseSelectorHook, useSelector } from 'react-redux';

import { currencyReducer, saga } from './ducks/currency';

const sagaMiddleware = createSagaMiddleware();
export const rootReducer = combineReducers({ currencyReducer });
const store = createStore(rootReducer, applyMiddleware(sagaMiddleware));
sagaMiddleware.run(saga);

export type RootState = ReturnType<typeof rootReducer>;
export const useTypedSelector: TypedUseSelectorHook<RootState>= useSelector;

export default store;
