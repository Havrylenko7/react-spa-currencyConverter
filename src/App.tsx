import React, { useEffect } from 'react';
import 'bootstrap/dist/css/bootstrap.min.css';
import { Routes, Route } from 'react-router-dom';
import { useDispatch } from 'react-redux';

import { getCurrencyFetch } from './redux/ducks/currency';
import { Navigation, Converter, Rates } from './components';
import { Grid, Loader } from './styled/App';
import { useTypedSelector } from './redux/store';

const App: React.FC = () => {
  const rates: {date: number, eur: []} = useTypedSelector(state => state.currencyReducer.rates);

  const dispatch = useDispatch();
  
  useEffect(() => {
    dispatch(getCurrencyFetch());
  }, []);
  
  if (Object.keys(rates)[0] !== 'date') {
    return <Loader>Loading...</Loader>
  } else {
    return (
      <Grid>
        <Navigation />
        <Routes>  
          <Route path="/rates" element={<Rates />}/>
          <Route path="/converter" element={<Converter />}/>
        </Routes> 
      </Grid>
    )
  };
};

export default App;
