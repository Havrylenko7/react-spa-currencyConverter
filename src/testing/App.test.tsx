import { Provider } from 'react-redux';
import { shallow, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import App from '../App';
import store from '../redux/store';

configure({ adapter: new Adapter() });

it('App should match a snapshot', () => {
  const app = shallow(<Provider store={store}><App /></Provider>)
  expect(app).toMatchSnapshot()
});
