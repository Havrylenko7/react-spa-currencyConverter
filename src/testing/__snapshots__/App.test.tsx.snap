// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `ShallowWrapper {}`;

exports[`App should match a snapshot 1`] = `ShallowWrapper {}`;

exports[`renders correctly 1`] = `ShallowWrapper {}`;
